### PR TITLE
Fix sensor card from not working with detail

### PR DESCRIPTION
**Pull request by zsarnett**
## Proposed change

I do not know how we left this for so long. The detail field was looking at `this._config.number` instead of `this._config.detail`. Like its been that way since creation.

image

My change makes this a switch. There are only two values this can be and I have 0 idea why it was made a number field in the first place

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/lovelace/editor/config-elements/hui-sensor-card-editor.ts [hunk 1]
```diff
@@ -152,28 +154,24 @@ export class HuiSensorCardEditor extends LitElement
...
-            )})"
-            type="number"
-            .value="${this._detail}"
-            .configValue="${"detail"}"
-            @value-changed="${this._valueChanged}"
+            .value=${this._unit}
+            .configValue=${"unit"}
+            @value-changed=${this._valueChanged}
           ></paper-input>
+          <ha-formfield label="Show more detail">
```
Comment (bramkragten) [inline src/panels/lovelace/editor/config-elements/hui-sensor-card-editor.ts:161]: Lokalise

File: src/panels/lovelace/editor/config-elements/hui-sensor-card-editor.ts [hunk 2]
```diff
@@ -182,15 +180,34 @@ export class HuiSensorCardEditor extends LitElement
...
+    const target = ev.target! as EditorTarget;
+    const value = target.checked ? 2 : 1;
+
+    if (this[`_${target.configValue}`] === value) {
+      return;
+    }
+
+    this._config = {
+      ...this._config,
+      [target.configValue!]: value,
```
Comment (bramkragten) [inline src/panels/lovelace/editor/config-elements/hui-sensor-card-editor.ts:206]: This is really specific for the detail option, so let's not make the rest generic

File: src/panels/lovelace/editor/config-elements/hui-sensor-card-editor.ts [hunk 3]
```diff
@@ -152,28 +154,24 @@ export class HuiSensorCardEditor extends LitElement
...
+            .value=${this._unit}
+            .configValue=${"unit"}
+            @value-changed=${this._valueChanged}
           ></paper-input>
+          <ha-formfield label="Show more detail">
+            <ha-switch
+              .checked=${this._detail === 2}
+              .configValue=${"detail"}
+              @change=${this._change}
+            ></ha-switch>
```
Comment (bramkragten) [inline src/panels/lovelace/editor/config-elements/hui-sensor-card-editor.ts:166]: Do we want this to be a switch or a slider?

Comment (zsarnett) [reply src/panels/lovelace/editor/config-elements/hui-sensor-card-editor.ts:166]: Slider? Its only 2 values

Comment (zsarnett) [reply src/panels/lovelace/editor/config-elements/hui-sensor-card-editor.ts:166]: Detail or no detail

Comment (bramkragten) [reply src/panels/lovelace/editor/config-elements/hui-sensor-card-editor.ts:166]: oh... why it is a number then? 😝

Comment (zsarnett) [reply src/panels/lovelace/editor/config-elements/hui-sensor-card-editor.ts:170]: no idea. Mentioned that above. 0 idea why this was made a number. I converted it to Typescript. Can't remember if I created this or not.

Comment (bramkragten) [reply src/panels/lovelace/editor/config-elements/hui-sensor-card-editor.ts:170]: Oh, should I read the description of the PR? 🙈

**Review by bramkragten — APPROVED** [review]

**Pull request merged**